experiment/simulation/js/fa.js

```diff
@@ -0,0 +1,333 @@
+import { registerGate } from "./main.js";
+import { setPosition } from "./layout.js";
+import {gates} from './gate.js';
+import {jsPlumbInstance} from "./main.js";
+
+// Dictionary of all full adders in the circuit with their IDs as keys
+let fullAdder = {}
+
+function clearFAs() {
+
+    for (let faID in fullAdder) {
+        delete fullAdder[faID];
+    }
+    fullAdder = {};
+}
+
+// {output-id: [gate,pos]}
+let finalOutputs = {
+    "Output-5": [],
+    "Output-8": [],
+    "Output-11": [],
+    "Output-12": [],
+};
+
+class FullAdder {
+    constructor() {
+        this.id = "FullAdder-" + window.numComponents++;

```

Comment:
Instead of using the global variable numComponents, can we pass the number as a constructor argument?

Reply:
No, because we still have to keep track of the total components created in our circuit to allocate unique ids, and since different components are created across different files, choosing global variables seemed to be the most convenient option.